assets/js/blocks/featured-product/block.js

```diff
@@ -28,178 +26,45 @@ import {
-			this.setState( { product: false, loaded: true, error: false } );
-			return;
-		}
-		apiFetch( {
-			path: `/wc/blocks/products/${ productId }`,
-		} )
-			.then( ( product ) => {
-				this.setState( { product, loaded: true, error: false } );
-			} )
-			.catch( ( apiError ) => {
-				const error = {
-					retry: this.debouncedGetProduct,
-				};
-
-				if ( isObject( apiError ) ) {
-					error.message = (
-						<span>
-							{ __( 'The following error was returned from the API', 'woo-gutenberg-products-block' ) }
-							<br />
-							<code>{ escapeHTML( apiError.message ) }</code>
-						</span>
-					);
-				} else {
-					error.message = __( 'An unknown error occurred which prevented the block from being updated.', 'woo-gutenberg-products-block' );
-				}
-
-				this.setState( { error: false } ); // Force update if error stays same.
-				this.setState( { product: false, loaded: true, error: error } );
-			} );
-	}
-
-	getInspectorControls() {
-		const {
-			attributes,
-			setAttributes,
-			overlayColor,
-			setOverlayColor,
-		} = this.props;
-
-		const url =
-			attributes.mediaSrc || getImageSrcFromProduct( this.state.product );
-		const { focalPoint = { x: 0.5, y: 0.5 } } = attributes;
-
-		return (
-			<InspectorControls key="inspector">
-				<PanelBody title={ __( 'Content', 'woo-gutenberg-products-block' ) }>
-					<ToggleControl
-						label={ __( 'Show description', 'woo-gutenberg-products-block' ) }
-						checked={ attributes.showDesc }
-						onChange={ () => setAttributes( { showDesc: ! attributes.showDesc } ) }
-					/>
-					<ToggleControl
-						label={ __( 'Show price', 'woo-gutenberg-products-block' ) }
-						checked={ attributes.showPrice }
-						onChange={ () => setAttributes( { showPrice: ! attributes.showPrice } ) }
-					/>
-				</PanelBody>
-				<PanelColorSettings
-					title={ __( 'Overlay', 'woo-gutenberg-products-block' ) }
-					colorSettings={ [
-						{
-							value: overlayColor.color,
-							onChange: setOverlayColor,
-							label: __( 'Overlay Color', 'woo-gutenberg-products-block' ),
-						},
-					] }
-				>
-					<RangeControl
-						label={ __( 'Background Opacity', 'woo-gutenberg-products-block' ) }
-						value={ attributes.dimRatio }
-						onChange={ ( ratio ) => setAttributes( { dimRatio: ratio } ) }
-						min={ 0 }
-						max={ 100 }
-						step={ 10 }
-					/>
-					{ !! FocalPointPicker && !! url &&
-						<FocalPointPicker
-							label={ __( 'Focal Point Picker' ) }
-							url={ url }
-							value={ focalPoint }
-							onChange={ ( value ) => setAttributes( { focalPoint: value } ) }
-						/>
-					}
-				</PanelColorSettings>
-			</InspectorControls>
-		);
-	}
-
-	renderEditMode() {
-		const { attributes, debouncedSpeak, setAttributes } = this.props;
+const FeaturedProduct = ( { attributes, debouncedSpeak, error, getProduct, isLoading, isSelected, overlayColor, product, setAttributes, setOverlayColor } ) => {
```

Comment:
Regarding `getProduct` I don't think the presentation component should receive any getter but instead only the values it uses for rendering.  I realize that it's being used as the `onRetry` method for `ApiErrorPlaceholder` but this to me smells like `ApiErrorPlaceholder` should be wrapped with the `withGetProduct` Hoc and exposed as `ProductApiErrorPlaceholder` which can then be used here.  The would remove the data coupling here.

Alternatively, in future refactors we could look at having error placeholders passed in as render props which would give more flexibility to code consuming the presentation components for how errors are handled.

Reply:
I have been investigating around this idea but I'm not sure to understand it yet. What would be the benefit of wrapping `ApiErrorPlaceholder` with `withProduct` if the parent component is already wrapped by that HOC? That means we are re-rendering the error component at least once just to get the `props` which were already loaded in its parent.

Am I missing something? I created a WIP here:

https://github.com/woocommerce/woocommerce-gutenberg-products-block/compare/fix/reset-loaded-state-in-featured-blocks...fix/reset-loaded-state-in-featured-blocks-error?expand=1